"Projects" and "Home Page" Edits | pins project filter to top of project list, un-sticky's the Filters on scroll, & centers components

agosmou opened the pull request:
Fixes #4593 

### What changes did you make and why did you make them ?

  - Pinned the projects filters to the top of the projects list to mimic the REI website
       - UX/UI best practice to facilitate user navigation along the page
  - Removed the "sticky" styling for the filters side pane
       - This caused unnecessary scrolling through all project cards to release the filters pane so the user can see the filters at the bottom of the pane
  - Centered the components to the center to mimic the REI website
     - having the components centered can be considered good UX/UI since the user can quickly navigate because there is less distance between components
- Refactored where needed 
 

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)
<!-- Note, if your images are too big, use the <img src="" width="" length="" />  syntax instead of ![image](link) to format the images -->
<!-- If images are not loading properly, you might need to double check the syntax or add a newline after the closing </summary> tag -->

<details>
<summary>Visuals before changes are applied</summary>

![image](https://github.com/hackforla/website/assets/73868258/290a0235-b877-4ca6-924d-2bf0787eee9d)

</details>

<details>
<summary>Visuals after changes are applied</summary>

![image](https://github.com/hackforla/website/assets/73868258/5f48a4a2-1972-4024-9d67-8cb35c124297)
  
![image](https://github.com/hackforla/website/assets/73868258/74b60e83-1cf8-48c7-a3ab-c457f0cf07e5)

</details>

agosmou committed:
removed sticky filter. Centered components on page. Moved filters to the top

agosmou committed:
removed unnessecary JS and adjusted the flex property for the filters

agosmou committed:
refactors

agosmou committed:
refactors html

t-will-gillis commented:
Availability: 5/18 evening, 5/19
ETA: 5/19

mademarc commented:
Review ETA: 5/18/2023
Availability: 7:05 PM

mademarc reviewed (dismissed):
Hey @agosmou  great adding the filter on the div class to keep the code very well read.

t-will-gillis reviewed (changes requested):
Hey @agosmou great job adjusting the projects filter: removing the sticky styling, pinning to the top of the project list, and centering on the page for the Projects and homepages. 

When I view in Docker, I am noticing the following:
- In all views, the 'Applied Filters' notice no longer appears. I am not arguing whether or not the 'Applied Filter' notice is useful or worth keeping, only that removing this notice is beyond the scope of the issue.
- In tablet and desktop views, the red applied filter tags have shifted from the Filters column and now are above the project cards. I believe that keeping the red applied filter tags in the current location is more intuitive for the user.
- In the tablet/ 767 px view, the Filters column width compresses and looks 'smushed' in comparison to the way it looks currently in the same view. I feel that maintaining the width of the Filters column is preferable visually.
![comparison_767](https://github.com/hackforla/website/assets/40799239/233cbfb7-bbf3-45f4-87fe-ac6c85ae1886)
- Lastly, I am not entirely clear about how the [/projects-check](https://www.hackforla.org/projects-check/) page is used currently or whether there is even a link to this page from the current website, however the edits made with the PR also change the layout of this page- both the 'Current Projects' header and the project cards themselves become off-center. 
![comparison_projects-check](https://github.com/hackforla/website/assets/40799239/4368eac3-1173-4fd9-bb80-ff4cb31e04f4)
Thanks for your work so far!

agosmou committed:
adds min-width to the filters side pane. Corrects projects check page

agosmou commented:
Hey @t-will-gillis - Thank you for your thoroughness. See Below:

***

> * In all views, the 'Applied Filters' notice no longer appears. I am not arguing whether or not the 'Applied Filter' notice is useful or worth keeping, only that removing this notice is beyond the scope of the issue.

I feel like this was implied as I had to match REI's page (where their filter's don't have that text) per [Bonnie's comment on the Emergent Request](https://github.com/hackforla/website/issues/4530#issuecomment-1521032707)

> * In tablet and desktop views, the red applied filter tags have shifted from the Filters column and now are above the project cards. I believe that keeping the red applied filter tags in the current location is more intuitive for the user.

I had to match REI/Amazon's page per [Bonnie's comment on the Emergent Request](https://github.com/hackforla/website/issues/4530#issuecomment-1521032707)

> * In the tablet/ 767 px view, the Filters column width compresses and looks 'smushed' in comparison to the way it looks currently in the same view. I feel that maintaining the width of the Filters column is preferable visually.

Agreed! I adjusted this.
![image](https://github.com/hackforla/website/assets/73868258/f2eb6f2d-da62-41ed-b85c-82ec25ef3ee4)


> * Lastly, I am not entirely clear about how the [/projects-check](https://www.hackforla.org/projects-check/) page is used currently or whether there is even a link to this page from the current website, however the edits made with the PR also change the layout of this page- both the 'Current Projects' header and the project cards themselves become off-center.

Nice catch - Those properties are used in a lot of places throughout the whole website. I went ahead and fixed this in the CSS just in case (not sure this page is used).
![image](https://github.com/hackforla/website/assets/73868258/b8c642d6-3003-40a3-a7e0-78b147cd9025)

t-will-gillis reviewed (approved):
Hey @agosmou  – Thanks for the clarification/ explanation about the placement of the red applied filter tags. I can see why you did what you did when I look at the REI page. Also, when the filter tags are at the top of the project cards, there is not really a good location for the ‘Applied Filters’ verbiage and since it does not add to clarity, I agree with you that the text can/should be removed.

<img width="977" alt="rei_comp" src="https://github.com/hackforla/website/assets/40799239/579b2be6-cc5c-4f92-857c-6d5d76317c5b">

Thanks also for changing the min width of the Filters column (I am just noticing the responsive design changes from 1 to 2 to 3 columns going from 766 to 767 to 768 on the current site) and for adjusting the /projects-check page. Both look good now.

Great job working through this issue!
